backend/fms_core/viewsets/dataset.py

```diff
@@ -0,0 +1,122 @@
+from datetime import datetime
+from django.http import HttpResponse, HttpResponseBadRequest
+from rest_framework import viewsets, status
+from rest_framework.response import Response
+from rest_framework.decorators import action
+from django.db.models import Exists, OuterRef, Q, Case, When, IntegerField, Max
+from django.core.exceptions import ValidationError
+from jsonschema.exceptions import ValidationError as JsonValidationError
+from fms_core.filters import DatasetFilter
+from fms_core.models.dataset import Dataset
+from fms_core.models.dataset_file import DatasetFile
+from fms_core.models._constants import ReleaseFlag
+from fms_core.serializers import DatasetFileSerializer, DatasetSerializer
+from fms_core.schema_validators import RUN_PROCESSING_VALIDATOR
+
+from ._utils import _list_keys
+from ._constants import _dataset_filterset_fields
+
+import fms_core.services.dataset as service
+
+class DatasetViewSet(viewsets.ModelViewSet):
+    queryset = Dataset.objects.all()
+    queryset = queryset.annotate(
+        last_release_timestamp=Max("files__release_flag_timestamp")
+    )
+
+    serializer_class = DatasetSerializer
+
+    ordering_fields = (
+        *_list_keys(_dataset_filterset_fields),
+        "last_release_timestamp"
+    )
+
+    filterset_fields = {
+        **_dataset_filterset_fields,
+    }
+
+    filter_class = DatasetFilter
+
+    @action(detail=False, methods=["post"])
+    def add_run_processing(self, request, *args, **kwargs):
+        data = request.data
+
+        def func(rpm):
```

Comment:
Is that function (func) really necessary ?

Reply:
it's mainly so i can return easily
maybe an exception would have been better...